### PR TITLE
Add tools that convert ftn files with WW3 switch pre-processing

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
# Pull Request Summary
Adds two files which can convert *.ftn to *.F90 files 

## Description
This PR adds two files to the model/tools directory which are used for converting files to src/*.F90 files with CPP switches from the original ftn/*.ftn files with WW3 switches.  This is the first step towards swtich->cpp conversion.  

Usage of the script, is from the model/tools directory: 
`sh ftn2src.sh`
This will convert your ftn/* folder which uses the switch preprocessing to CPP if defs.  Note a prefix is added to <WW3SWITCH>  to be defined as the CPP switch:  W3_<WW3SWITCH>  in the code.  This runs the script switch2cpp.awk written by @ukmo-ccbunney three times, so all double switches will be processed. 

### Issue(s) addressed
* This is the first step towards #360

### Check list  
* Is your feature branch up to date with the authoritative repository (NOAA/develop)?
* Make sure you have checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop) and [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number)
* Please list appropriate labels code managers should add for this PR:   
 _bug_, _documentation_, _enhancement_, _new feature_, ..
 
* Reviewers: @mentions of suggested reviewers of the proposed changes

### Commit Message
Add tools that convert ftn files with WW3 switch pre-processing to src/*.F90 files with CPP switches
* This is the first step towards swtich->cpp conversion
* To use this script, from the tool directory run sh ftn2src.sh
* This processes the code 3 times so double switches are accounted for

### Testing
* How were these changes tested?These scripts are tested in subsequent PRs but are just adding the two routines here and therefore are not tested within this PR.  

Co-author: @ukmo-ccbunney

### CODE REVIEW

File: model/tools/ftn2src.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  ftnfiles=$(sed -e "s/.ftn//g" <<< $ftnfiles)
+  nonftnfiles=`ls -I "*.ftn"`
+ 
+  cd $thisdir
+  if [ ! -d $srcdir/$DIR ] 
+  then 
+     mkdir -p $srcdir/$DIR
+  fi  
+
+  set -x
```
Comment (ukmo-ccbunney) [inline model/tools/ftn2src.sh]: `set -x` 
Can we remove this? It makes it very verbose.

**Review by ukmo-ccbunney — CHANGES_REQUESTED** [review]
Works fine for me! Just a few teeny tiny changes, if you agree...

File: model/tools/ftn2src.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
+
```
Comment (ukmo-ccbunney) [inline model/tools/ftn2src.sh:2]: Do we want a nice shebang at the top? E.g.
`!/bin/bash`

File: model/tools/ftn2src.sh (hunk 1 above)
Comment (JessicaMeixner-NOAA) [reply model/tools/ftn2src.sh]: Yes - I forgot to remove it after debugging some things.

File: model/tools/ftn2src.sh (hunk 2 above)
Comment (JessicaMeixner-NOAA) [reply model/tools/ftn2src.sh:2]: Smart idea! I'll add this.

**Comment by aliabdolali** [comment]
@JessicaMeixner-NOAA is this ready to be tested? Can you provide a description on the wiki page for the users?

**Comment by JessicaMeixner-NOAA** [comment]
Actually instead of a wiki -- why don't I add information in a README in the tools section or to the bin director readme?   This really should be "tested" in the other PRs.  I'll get everything ready here shortly as the other PRs have been merged now.  In the meantime, should we address the Expanse porting PR?

**Comment by aliabdolali** [comment]
> Actually instead of a wiki -- why don't I add information in a README in the tools section or to the bin director readme? This really should be "tested" in the other PRs. I'll get everything ready here shortly as the other PRs have been merged now. In the meantime, should we address the Expanse porting PR?

Yup, README is ok with me. 
 Expanse porting PR is running now. I fixed the conflicts and submitted the tests. It will be ready today.

**Comment by aliabdolali** [comment]
For this PR, I would follow the instruction and convert *ftn files to *F90 (3times) and will make sure no switch is left. There is no need to run regtests.

**Review by aliabdolali — APPROVED** [review]

**Pull request merged**